distribution/integration-tests/src/test/java/org/jboss/tools/rsp/itests/wildfly/QuickstartsDeploymentTest.java

```diff
@@ -0,0 +1,235 @@
+	}
+
+	@After
+	public void after() throws Exception {
+		stopServer(client, SERVER_ID);
+		deleteServer(SERVER_ID);
+	}
+
+	@Test
+	public void testAppDeploymentToServer() throws Exception {
+		// Verify project path
+		assertTrue("File " + projectPath + " does not exist", Files.exists(projectPath, LinkOption.NOFOLLOW_LINKS));
+		// No deployable avaiable at the moment
+		assertTrue(serverProxy.getDeployables(handle).get().isEmpty());
+		serverProxy.addDeployable(new ServerDeployableReference(handle, reference));
+		waitForDeployablePublishState(ServerManagementAPIConstants.PUBLISH_STATE_ADD, 10, client);
+		verifyURL(url.toString(), 404);
+
+		// Deployable available, but not deployed yet until publish full request send to
+		// the server
+		DeployableState state = getDeployableStates(handle, reference);
+		assertNotNull("Deployable reference was not found", state);
+		assertDeployableReference(state);
+		verifyURL(url.toString(), 404);
+
+		// send publish request with publish full kind
+		PublishServerRequest pubReq = new PublishServerRequest(handle, ServerManagementAPIConstants.PUBLISH_FULL);
+		Status statusPublish = serverProxy.publish(pubReq).get();
+		assertEquals("Publish request status OK check failed, was: " + statusPublish.getMessage(), Status.OK,
+				statusPublish.getSeverity());
+
+		// check for the publishing is done, then it requires no publish change
+		waitForDeployablePublishState(ServerManagementAPIConstants.PUBLISH_STATE_NONE, 10, client);
+		state = getDeployableStates(handle, reference);
+		assertNotNull("Deployable reference was not found", state);
+		assertDeployableReference(state);
+		// even though publishing is done, app is not yet deployed, thus waiting takes
+		// place
+		HttpUtility.waitForUrlEndpoint(url, 200, 15);
+		verifyURL(url.toString(), 200);
+
+		// Remove deployment
+		Status statusRemove = serverProxy.removeDeployable(new ServerDeployableReference(handle, reference)).get();
+		assertEquals("Publish request status OK check failed, was: " + statusRemove.getMessage(), Status.OK,
+				statusRemove.getSeverity());
+		state = getDeployableStates(handle, reference);
+		// Deployable state of the depl. is still present
+		assertNotNull("Deployable reference was not found", state);
+		assertDeployableReference(state);
+		verifyURL(url.toString(), 200);
+
+		// Another publish full request must be sent
+		statusPublish = serverProxy.publish(pubReq).get();
+		assertEquals("Publish request status OK check failed, was: " + statusPublish.getMessage(), Status.OK,
+				statusPublish.getSeverity());
+		// Deployment is no more
+		assertTrue(serverProxy.getDeployables(handle).get().isEmpty());
+		state = getDeployableStates(handle, reference);
+		assertNull("Deployable reference was still found", state);
+		HttpUtility.waitForUrlEndpoint(url, 404, 10);
+		verifyURL(url.toString(), 404);
+	}
+
+	protected DeployableState getDeployableStates(ServerHandle handle, DeployableReference reference) {
```

Comment:
This should be private, or moved to the superclass if needed.
Also the plural in the name sort of suggests it returns multiple states, not just one.